docs/reference/query-dsl/term-level-queries.asciidoc

```diff
@@ -3,7 +3,8 @@
 
 While the <<full-text-queries,full text queries>> will analyze the query
 string before executing, the _term-level queries_ operate on the exact terms
-that are stored in the inverted index.
+that are stored in the inverted index, except for keyword fields with <<normalizer,normalizer>> property, 
```

Comment:
"except" here might sound like term level queries operate on something other than terms that are stored in the inverted index, while this is actually the case. The input for keyword fields will just be normalized before storing it, and so will the query. 
I understand the desire to clarify this but I am on the fence of doing it here, since the purpose of this paragraph is mostly to list all the term queries, not to go into details about the analysis/normalization process. I'm wondering if there is another place in the docs where this might fit better. Alternatively, could you rephrase it so it reflects what I mentioned before, but in a very brief way so it doesn't distract much from the main purpose of this section?

Reply:
Thank you @cbuescher for your review :+1: 
I agree that this page is mostly to list only all the term queries. But the first line is discussing about **full text queries that analyze query string before executing**, and that term level queries operate on **the exact terms**: I think this is  wrong, because term level queries will normalize terms string before executing (in keyword field case) and will not operate on the exact terms.

This is an example (with elasticsearch 6.2.4):

Create an index with one normalized field (use only filter lowercase):

```
PUT localhost:9200/coutries

{
  "settings": {
    "analysis": {
      "normalizer": {
        "my_normalizer": {
          "type": "custom",
          "char_filter": [],
          "filter": ["lowercase"]
        }
      }
    }
  },
  "mappings": {
    "country": {
      "properties": {
        "name": {
          "type": "keyword",
          "index": true,
          "normalizer": "my_normalizer"
        }
      }
    }
  }
}

```

Insert some documents:

```
POST localhost:9200/coutries/country/_bulk

{ "index" : { "_id" : "1" } }
{ "name" : "Netherlands" }
{ "index" : { "_id" : "2" } }
{ "name" : "Tunisia" }
{ "index" : { "_id" : "3" } }
{ "name" : "France" }
```

And this is what I get when I use term levels queries with different way (UPPERCASE, lowercase, mixed):

- **Term query**

```
POST localhost:9200/coutries/country/_search

{
  "query": {
    "term" : { "name" : "NETHERLANDS" } 
  }
}

=> Response
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 0.2876821,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "1",
        "_score": 0.2876821,
        "_source": {
          "name": "Netherlands"
        }
      }
    ]
  }
}
```

- **Terms query**

```
POST localhost:9200/coutries/country/_search

{
    "query": {
        "terms" : { "name" : ["netherlands", "TUNISIA", "France"]}
    }
}

=> Response
{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 3,
    "max_score": 1,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "2",
        "_score": 1,
        "_source": {
          "name": "Tunisia"
        }
      },
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "1",
        "_score": 1,
        "_source": {
          "name": "Netherlands"
        }
      },
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "3",
        "_score": 1,
        "_source": {
          "name": "France"
        }
      }
    ]
  }
}

```

- **Prefix query**

```
POST localhost:9200/coutries/country/_search

{ "query": {
    "prefix" : { "name" : "tun" }
  }
}

=> Response
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 1,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "2",
        "_score": 1,
        "_source": {
          "name": "Tunisia"
        }
      }
    ]
  }
}
```

- **Wildcard query**

```
POST localhost:9200/coutries/country/_search

{
    "query": {
        "wildcard" : { "name" : "f*CE" }
    }
}

=> Response
{
  "took": 5,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 1,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "3",
        "_score": 1,
        "_source": {
          "name": "France"
        }
      }
    ]
  }
}
```

So as we can see here,  term level queries will not operate on the exact terms but will normalize terms string before executing for keyword field.

What I suggest is:

- rephrase with this way (maybe you have a better suggestion): 

```
While the <<full-text-queries,full text queries>> will analyze the query string before executing, 
the _term-level queries_ operate on the exact terms that are stored in the inverted index, and 
will normalize terms for keyword fields with <<normalizer,normalizer>> property.
```

- modify normalizer documentation page too: https://www.elastic.co/guide/en/elasticsearch/reference/master/normalizer.html

Instead of:

`The normalizer is applied prior to indexing the keyword, as well as at search-time when the keyword field is searched via a query parser such as the match query.`

We can write:

`The normalizer is applied prior to indexing the keyword, as well as at search-time.`

Or 

`The normalizer is applied prior to indexing the keyword, as well as at search-time when the keyword field is searched via a query parser such as the match query or via a term level queries such as the term query.`


What do you think @cbuescher ? (I will rephrase, format and submit changes after this messages exchange)